php/class-relate.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Relate class for the Cloudinary plugin.
+ *
+ * @package Cloudinary
+ */
+
+namespace Cloudinary;
+
+use Cloudinary\Connect\Api;
+use Cloudinary\Relate\Relationship;
+use WP_Query;
+
+/**
+ * Class Relate
+ */
+class Relate {
+
+	/**
+	 * Holds the plugin instance.
+	 *
+	 * @var Plugin Instance of the global plugin.
+	 */
+	public $plugin;
+
+	/**
+	 * Holds the Media instance.
+	 *
+	 * @var Media
+	 */
+	protected $media;
+
+	/**
+	 * Relate constructor.
+	 *
+	 * @param Plugin $plugin Instance of the main plugin.
+	 */
+	public function __construct( $plugin ) {
+		$this->plugin = $plugin;
+		$this->media  = $plugin->get_component( 'media' );
+		$this->register_hooks();
+	}
+
+	/**
+	 * Register hooks.
+	 */
+	protected function register_hooks() {
+		add_action( 'cloudinary_upgrade_asset', array( $this, 'upgrade_relation' ), 10, 2 );
+		add_action( 'found_posts', array( $this, 'warm_cache' ), 10, 2 );
+	}
+
+	/**
+	 * Warm the cache for found posts.
+	 *
+	 * @param int      $found_posts The number of posts found.
+	 * @param WP_Query $query       The WP_Query instance (passed by reference).
+	 */
+	public function warm_cache( $found_posts, $query ) {
+		if ( ! empty( $found_posts ) && 'attachment' === $query->query_vars['post_type'] ) {
+			Relationship::preload( $query->posts );
+		}
+	}
+
+	/**
+	 * Upgrade an asset relation.
+	 *
+	 * @param int    $attachment_id The attachment ID.
+	 * @param string $version       The version upgrading to.
+	 */
+	public function upgrade_relation( $attachment_id, $version ) {
+		$asset_plugin_version = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['plugin_version'], true );
+		if ( ! empty( $asset_plugin_version ) && version_compare( $asset_plugin_version, $version, '<' ) ) {
+			$transformations = $this->media->get_post_meta( $attachment_id, Sync::META_KEYS['transformation'], true );
```

Comment:
@DavidCramer 

I'm sorry, I should had comment here instead of https://github.com/cloudinary/cloudinary_wordpress/pull/773#discussion_r916072103